atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/keyvalue/api/cache/ValueStoreImplTest.java

```diff
@@ -103,28 +113,45 @@ public void watchEventUpdatesWatchableTables() {
     @Test
     public void valuesEvictedOnceMaxSizeReached() {
         // size is in bytes; with overhead, this should keep 2 but not three values
-        valueStore = new ValueStoreImpl(ImmutableSet.of(TABLE), 300);
+        valueStore = new ValueStoreImpl(ImmutableSet.of(TABLE), 300, metrics);
         CellReference tableCell2 = CellReference.of(TABLE, CELL_2);
 
         valueStore.applyEvent(WATCH_EVENT);
         valueStore.putValue(TABLE_CELL, VALUE_1);
         valueStore.putValue(tableCell2, VALUE_2);
+        verify(metrics, times(2)).increaseCacheSize(anyLong());
+
         valueStore.putValue(CellReference.of(TABLE, CELL_3), VALUE_3);
+        verify(metrics, times(3)).increaseCacheSize(anyLong());
+        verify(metrics).decreaseCacheSize(anyLong());
 
         // Caffeine explicitly does *not* implement simple LRU, so we cannot reason on the actual entries here.
         assertThat(((ValueCacheSnapshotImpl) valueStore.getSnapshot()).values()).hasSize(2);
     }
 
     @Test
     public void lockedValuesDoNotCountToCacheSize() {
-        valueStore = new ValueStoreImpl(ImmutableSet.of(TABLE), 300);
+        valueStore = new ValueStoreImpl(ImmutableSet.of(TABLE), 300, metrics);
         valueStore.applyEvent(WATCH_EVENT);
         valueStore.applyEvent(LOCK_EVENT);
 
         valueStore.putValue(CellReference.of(TABLE, CELL_2), VALUE_2);
         valueStore.putValue(CellReference.of(TABLE, CELL_3), VALUE_3);
         assertExpectedValue(CELL_2, CacheEntry.unlocked(VALUE_2));
         assertExpectedValue(CELL_3, CacheEntry.unlocked(VALUE_3));
+
+        verify(metrics, times(2)).increaseCacheSize(anyLong());
+    }
+
+    @Test
+    public void metricsCorrectlyCountOverlappingPuts() {
+        valueStore = new ValueStoreImpl(ImmutableSet.of(TABLE), 300, metrics);
+        valueStore.applyEvent(WATCH_EVENT);
+
+        valueStore.putValue(CellReference.of(TABLE, CELL_2), VALUE_2);
+        valueStore.putValue(CellReference.of(TABLE, CELL_2), VALUE_2);
+        verify(metrics, times(2)).increaseCacheSize(anyLong());
+        verify(metrics).decreaseCacheSize(anyLong());
```

Comment:
Throughout this test: would it make sense to assign values rather than using `anyLong()` throughout?

Reply:
Done, although it leaks implementation details a bit